src/Groups/GAPGroups.jl

```diff
@@ -0,0 +1,617 @@
+The parent of `x` is set as Sym(n), where n is the largest moved point of `x`.
+"""
+function perm(L::Array{Int64,1})
+   return PermGroupElem(symmetric_group(length(L)), GAP.Globals.PermList(GAP.julia_to_gap(L)))
+end
+
+"""
+    perm(g::PermGroup, L::Array{Int64,1})
+Given the array of positive integers `V`, returns the permutation `x` sending V[i] into V[i+1] for every i. V must contain exactly one time every integer from 1 to n for n = length(V).
+The parent of `x` is g. If `x` is not in g, it return ERROR.
+"""
+function perm(g::PermGroup, L::Array{Int64,1})
+   x = GAP.Globals.PermList(GAP.julia_to_gap(L))
+   if GAP.Globals.IN(x,g.X) 
+     return PermGroupElem(g, GAP.Globals.PermList(GAP.julia_to_gap(L)))
+   end
+   throw(ArgumentError("the element does not embed in the group"))
+end
+
+# cperm stays for "cycle permutation", but we can change name if we want
+# takes as input a list of arrays (not necessarly disjoint)
+"""
+    cperm(L::Union{Array{Int64,1},UnitRange{Int64}}...)
+For given lists of positive integers `[a_1, a_2, ..., a_n],[b_1, b_2, ... , b_m], ...` return the
+permutation `x = (a_1,a_2,...,a_n)(b_1,b_2,...,b_m)...`. The array `[n,n+1,...,n+k]` can be replaced by `n:n+k`.
+  
+If a list is epmty or contains duplicates or holes, it fails.
+The parent of `x` is set as Sym(n), where n is the largest moved point of `x`.
+"""
+function cperm(L::Union{Array{Int64,1},UnitRange{Int64}}...)
+   if length(L)==0
+      return one(symmetric_group(1))
+   else
+      return prod([PermGroupElem(symmetric_group(maximum(y)), GAP.Globals.CycleFromList(GAP.julia_to_gap(collect(y)))) for y in L])
+   end
+end
+
+# cperm stays for "cycle permutation", but we can change name if we want
+# takes as input a list of arrays (not necessarly disjoint)
+# WARNING: we allow e.g. PermList([2,3,1,4,5,6]) in Sym(3)
+
+"""
+    cperm(G::PermGroup, L::Union{Array{Int64,1},UnitRange{Int64}}...)
+For given lists of positive integers `[a_1, a_2, ..., a_n],[b_1, b_2, ... , b_m], ...` return the
+permutation `(a_1,a_2,...,a_n)(b_1,b_2,...,b_m)...`. The array `[n,n+1,...,n+k]` can be replaced by `n:n+k`.
+  
+If a list is epmty or contains duplicates or holes, it fails.
+The parent of `x` is G. If `x` is not in G, it return ERROR.
+"""
+function cperm(g::PermGroup,L::Union{Array{Int64,1},UnitRange{Int64}}...)
+   if length(L)==0
+      return one(g)
+   else
+      x=GAP.Globals.Product(GAP.julia_to_gap([GAP.Globals.CycleFromList(GAP.julia_to_gap(collect(y))) for y in L]))
+      if GAP.Globals.IN(x,g.X) return PermGroupElem(g, x)
+      else throw(ArgumentError("the element does not embed in the group"))
+      end
+   end
+end
+
+"""
+    listperm(x::PermGroupElem)
+is the list L defined by L = [ `x`(i) for i in 1:n ], where Sym(n) = parent(`x`).
+"""
+function listperm(x::PermGroupElem)
```

Comment:
`collect` would be better, but it's a question if we want to be able to iterate over `PermGroupElem

Reply:
Why would `collect` be better?! Looking at its documentation:
> Return an Array of all items in a collection or iterator
A permutation is neither a collection nor an iterator nor iterable. I'd find it rather confusing. But perhaps I am missing something -- what's your reasoning for considering `collect` better?

Reply:
because a permutation is a sequence (an ordered collection?) of values indexed by natural numbers; even though it doesn't implement iterator protocol. But I'm just suggesting a different name; If you prefer function-based approach `images` might be better; there are no `lists` in julia, so unless you know the function it'll be very hard to find, that's all;

Reply:
Corrected it. But I don't see why should we collect over `PermGroupElem`, this is just a permutation, not any kind of set.

Reply:
that's correlation, not causation ;)